modules/data.atmosphere/R/temporal.downscaling.R

```diff
@@ -100,85 +104,116 @@ cfmet.downscale.subdaily <- function(subdailymet, output.dt = 1) {
     }
   }
 
-  downscaled.result <- cbind(new.date, data.table::as.data.table(downscaled.result))
+  downscaled.result <- cbind(new.date, as.data.frame(downscaled.result))
 } # cfmet.downscale.subdaily
 
 
+#' Internal helper to downscale a single row from a daily file
+#'
+#' @param df one row from dailymet
+#' @param tseq vector of hours at which to estimate
+#' @param lat latitude
+#'
+#' @return df with one row for each hour in `tseq`
+#'
+downscale_one_cfmet_day <- function(df, tseq, lat) {
+  if (nrow(df) != 1) {
+    PEcAn.logger::logger.error("df must be a one-row dataframe")
+  }
+  if (length(unique(diff(tseq))) != 1) {
+    PEcAn.logger::logger.error("tseq has uneven intervals")
+  }
+
+  n <- length(tseq)
+
+  light <- lightME(DOY = df$doy, t.d = tseq, lat = lat) %>%
+    as.data.frame() %>%
+    dplyr::mutate(
+      Itot = .data$I.dir + .data$I.diff,
+      resC2 = (.data$Itot - min(.data$Itot)) / max(.data$Itot))
+
+  rhscale <- (cos(2 * pi * (tseq - 10) / n) + 1) / 2
+
+  data.frame(
+    hour = tseq,
+    solarR = rep(
+        df$surface_downwelling_shortwave_flux_in_air * 2.07 * 10^5 / 36000,
+        each = n)
+      * light$resC2,
+    # Note: When dt >= 6, all times get the same T prediction
+    Temp = df$tmin +
+      (sin(2 * pi * (tseq - 10) / n) + 1) /
+      2 * (df$tmax - df$tmin),
+    relative_humidity = df$rhmin + rhscale * (df$rhmax - df$rhmin),
+    # TODO: Why do we divide by n?
+    #   isn't precipitation_flux already an intensity?
+    precipitation_flux = rep(df$precipitation_flux / n, each = n),
```

Comment:
The CF units on `precipitation_flux` are documented as kg m^-2 s^-1, so I would have expected it to be an average rate across the day and therefore to not need dividing by `n`. Is the division incorrect or am I missing something else? Is there a chance this input quantity was supposed to be named `precipitation_amount` (with units kg m^-2 = mm)?